test/transformer/transformer.test.ts

```diff
@@ -0,0 +1,59 @@
+import {Transformer} from '../../src/transformer';
+import {ConfigVersionDocument} from '../utils/version-document';
+
+const transformer = new Transformer();
+
+describe('Transformer', () => {
+  it('emits a key/value map of environment names to config objects', () => {
+    const res = transformer.transform('0.0.1', ConfigVersionDocument.VALID);
+    expect(res.production.android.platform).toEqual('Android');
+    expect(res.production.ios.platform).toEqual('iOS');
+  });
```

Comment:
Just a stylistic thing, but it might be better to test the Android and iOS transforms separately. I can see tests that only target Android or iOS. It would be much clearer to cleanly separate them out - if they truly differ - I think. If they don't differ then running all tests in a loop (one for Android and one for iOS) would be acceptable here.

Reply:
i went with this approach because the duality allows us to see the concept of 'overriding' in action - which is really the meat of this whole thing. so it allows you to say 'A gets overridden for this particular field, but B remains the same' - the fact it's iOS and Android are really just placeholders for A and B here, the logic isn't testing anything intrinsic to one platform or the other

Reply:
Similar to my comment about the version number, if iOS and Android are meaningless in this context - why not use something either random or totally made up? Popular sources for `made up stuff` at GDS include Harry Potter, LOR, H2G2 and extra points from some quarters for Star Wars.

Reply:
yes could do but would necessitate injecting the values of the 'platforms' and then updating the config version document to use them (though i do like the idea of calling `const transformer = new ConfigTransformer({ platforms: ['gryffindor', 'slytherin'] });`) - but feel like it might be more effort than it's worth - thoughts?